### PR TITLE
Adding simple config guide

### PR DESCRIPTION
**Pull request by samolego**
I've added a guide to creating simple configs using `gson`. I plan to extend it further as well

What do you think about it?

### CODE REVIEW

**Comment by Patbox** [comment]
Looks good to me

**Comment by samolego** [comment]
Sorry for last two small commits, it seems that my checks weren't good 😄 ...

**Comment by Southpaw1496** [comment]
I wouldn't worry, they'll probably be squashed in the end.

**Review by Tom-The-Geek — APPROVED** [review]
Seems good to me :thumbsup:

File: docs/config/gson_config.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+See below example.
+
+```java
+public class MyModConfig {
+
+    public float activationRange = 8.0F;
+
+    public boolean show = true;
+
+	public String message = "This is a config guide.";
```
Comment (Genau6502) [inline docs/config/gson_config.md]: Indentation

File: docs/config/gson_config.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+}
+```
+
+## Saving, reading and parsing
+
+If `config` only exists in our memory, it's not that usefull, as changes will be reset
+after server restart. If we want to make them permanent, we should write the config to a file.
+
+```java
+public class MyModConfig {
```
Comment (Genau6502) [inline docs/config/gson_config.md:57]: Could GSON handle this being a record?

File: docs/config/gson_config.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+
+Now that we have those new methods, we should use the following code to load the config.
+```java
+File configFile = new File("./config/my_mod_config.json");
```
Comment (Genau6502) [inline docs/config/gson_config.md]: `FabricLoader.getInstance().getConfigDir().resolve(“my_mod_config.json”);`

**Review by Genau6502 — CHANGES_REQUESTED** [review]

File: docs/config/gson_config.md (hunk 2 above)
Comment (Patbox) [reply docs/config/gson_config.md:57]: I'm sure gson (at least one bundled with Minecraft) errors with them

File: docs/config/gson_config.md (hunk 1 above)
Comment (samolego) [reply docs/config/gson_config.md]: Replaced tabs with spaces now :)

File: docs/config/gson_config.md (hunk 2 above)
Comment (samolego) [reply docs/config/gson_config.md:57]: Seems like that it's not implemented yet; https://github.com/google/gson/issues/1794.

File: docs/config/gson_config.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
+```
+
+Now that we have those new methods, we should use the following code to load the config.
+```java
+File configFile = FabricLoader.getInstance().getConfigDir().resolve(“my_mod_config.json”);
```
Comment (Genau6502) [inline docs/config/gson_config.md]: Apologies for my mobile formatting, could these be the non-fancy type of quotes?

Comment (samolego) [reply docs/config/gson_config.md]: Ah, sure

File: docs/config/gson_config.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+    public String message = "This is a config guide.";
+    
+    public NestedValues nested = new NestedValues();
+    
+    public static class NestedValues {
+        public String message = "This is a another message.";
+    }
+    
+    public ArrayList<String> randomQuestions = new ArrayList<>(Arrays.asList(
```
Comment (Tom-The-Geek) [inline docs/config/gson_config.md]:
```suggestion
    public List<String> randomQuestions = new ArrayList<>(Arrays.asList(
```
Usually when storing lists in a field you don't care about the implementation type

**Review by Tom-The-Geek — CHANGES_REQUESTED** [review]

File: docs/config/gson_config.md (hunk 5 above)
Comment (samolego) [reply docs/config/gson_config.md]: Thanks, wasn't aware of that :)

**Pull request merged**